### PR TITLE
Resolves #490

### LINKED ISSUE #490: Add new certificate endpoints

### PR DESCRIPTION
**Pull request by fradinni**

### CODE REVIEW

File: src/sdk/model/edge/certificates/__json__/crl-json.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { X509CRLJson } from './x509-crl-json';
+
+export interface CRLJson {
+  object_id: string;
+  description: string;
+  scope: string;
+  is_temporal: boolean;
+  issuer_cn: string;
+  pem_encoding: string;
+  x509_c_r_l: X509CRLJson;
```
Comment (JoMas971) [inline src/sdk/model/edge/certificates/__json__/crl-json.ts]: i think this is a typo error isn't it ? It should be `x509_crl`.

Comment (fradinni) [reply src/sdk/model/edge/certificates/__json__/crl-json.ts]: @JoMas971 Field is returned like that from API, it has to be updated core side.

Comment (JoMas971) [reply src/sdk/model/edge/certificates/__json__/crl-json.ts]: @fradinni yeah, in those cases you need to tell the core team to let them update on their side. 
It's a JSON format issue core side ;-).

Comment (fradinni) [reply src/sdk/model/edge/certificates/__json__/crl-json.ts]: @JoMas971  Good catch or the permission,  just added it.

Comment (cfsnyder) [reply src/sdk/model/edge/certificates/__json__/crl-json.ts]: indeed, @JoMas971  is also right about the format of that JSON parameter being a bug. Please let Collin know when you see issues like that so he can fix them. I'll tell him about this one.

File: src/sdk/model/edge/certificates/certificate-create-request.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      // Json or empty constructor
+      this._json = (firstParam || {}) as CertificateCreateRequestJson;
+    }
+  }
+
+  /**
+   * Get trust object.
+   * @returns {TrustObjectRequestJson}
+   */
+  get trustObject(): TrustObjectRequestJson {
```
Comment (cfsnyder) [inline src/sdk/model/edge/certificates/certificate-create-request.ts]: this should return the wrapper object vs. JSON representation

File: src/sdk/model/edge/certificates/csr-subject.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+   */
+  get c(): string {
+    return this._json.c;
+  }
+
+  /**
+   * Convert to an Array of Attributes.
+   * @returns {Array<Attribute>}
+   */
+  toAttributeArray(): Array<Attribute> {
```
Comment (cfsnyder) [inline src/sdk/model/edge/certificates/csr-subject.ts]: nice :+1:

File: src/sdk/model/edge/certificates/__json__/crl-json.ts (hunk 1 above)
Comment (fradinni) [reply src/sdk/model/edge/certificates/__json__/crl-json.ts]: @JoMas971  yeah i know. I noticed it when john pointed it out, i haven't seen this one was misformatted. Thanks @cfsnyder .

File: src/sdk/model/edge/certificates/certificate-create-request.ts (hunk 2 above)
Comment (fradinni) [reply src/sdk/model/edge/certificates/certificate-create-request.ts]: Fixed

**Pull request merged**